src/docfx/lib/schema/JsonSchemaValidator.cs

```diff
@@ -17,34 +18,43 @@ internal class JsonSchemaValidator
         private readonly JsonSchema _schema;
         private readonly JsonSchemaDefinition _definitions;
         private readonly MicrosoftGraphAccessor? _microsoftGraphAccessor;
-        private readonly ListBuilder<(JsonSchema schema, string key, JToken value, SourceInfo? source)> _metadataBuilder;
+        private readonly ListBuilder<(JsonSchema schema, string key, JToken value, SourceInfo? source, bool? isCanonicalVersion)> _metadataBuilder;
+        private static readonly ThreadLocal<bool?> _isCanonicalVersion = new ThreadLocal<bool?>();
```

Comment:
nit: use `t_` for thread local variables to tell the different between member variables and thread static variables.
```suggestion
        private static readonly ThreadLocal<bool?> t_isCanonicalVersion = new ThreadLocal<bool?>();
```